### DISCUSSION TITLE
Example.conf from 4.0.0 tutorial doesn't work in 4.1.0

### DISCUSSION THREAD

**Discussion by ksearight** [dtcenter/METplus#1536] [Configuration] [answered]
I've installed METplus 4.1.0 on AWS and am going through the 4.0.0 tutorial.  After setting up the environment, I run this command: ```
run_metplus.py \   
${METPLUS_BUILD_BASE}/parm/use_cases/met_tool_wrapper/Example/Example.conf
```.  The result is this error:  `configparser.NoOptionError: No option ‘OMP_NUM_THREADS’ in section: ‘config’`.   
  
I'm able to get past the error by adding `OMP_NUM_THREADS = 1` to the `config` section of `Example.conf`.  Is that a correct workaround?  I'm also wondering if there may be other changes in 4.1.0 that may cause problems when going through the 4.0.0. tutorial.

**Comment by georgemccabe** [comment]
It looks like you are in a weird state where you are running the run_metplus.py script from 4.1.0 but you are using the default parameters from 4.0.0. OMP_NUM_THREADS should be set in the defaults.conf. This would happen if you have the environment variable METPLUS_PARM_BASE set to the parm directory of the 4.0.0 install. This environment variable was used in the 3.1 tutorial but should not be used in the 4.0.0 tutorial. Unsetting that variable should fix this issue.

**Reply by ksearight** [reply]
I'm using the 4.1.0 install and don't have METPLUS_PARM_BASE set.

**Reply by georgemccabe** [reply]
Reviewing the first few lines log file should help you see what is being used. I recreated this error on my machine and this is what was shown in my log file. Note the bold text:

> 
> 03/23 17:41:45.123 metplus (config_metplus.py:105) DEBUG: Completed METplus configuration setup.
> 03/23 17:41:45.123 metplus INFO: Running METplus v4.1.0 called with command: /d1/personal/mccabe/METplus-4.1.0/ush/run_metplus.py /d1/personal/mccabe/METplus-4.1.0/parm/use_cases/met_tool_wrapper/Example/Example.conf /home/mccabe/mccabe.seneca.conf
> 03/23 17:41:45.123 metplus INFO: Log file: /d1/personal/mccabe/out/logs/metplus.log.20220323174145
> 03/23 17:41:45.123 metplus INFO: **METplus Base: /d1/personal/mccabe/METplus-4.1.0**
> 03/23 17:41:45.123 metplus INFO: Final Conf: /d1/personal/mccabe/out/metplus_final.conf
> 03/23 17:41:45.123 metplus INFO: Config Input: /d1/personal/mccabe/**METplus-4.0.0/parm/metplus_config/defaults.conf**
> 03/23 17:41:45.123 metplus INFO: Config Input: /d1/personal/mccabe/METplus-4.1.0/parm/use_cases/met_tool_wrapper/Example/Example.conf
> ...
> configparser.NoOptionError: No option 'OMP_NUM_THREADS' in section: 'config'

In my case the defaults.conf file is from 4.0.0 because I set METPLUS_PARM_BASE. If the defaults.conf file is from the 4.1.0 install and it does not include OMP_NUM_THREADS (it should be the variable after PROCESS_LIST), then perhaps that file was copied from the 4.0.0 version.

**Reply by ksearight** [reply]
Here's what my log file has for the same lines:
```
03/22 23:13:14.213 metplus INFO: Running METplus v4.1.0 called with command: /home/centos/METplus/ush/run_metplus.py /home/centos/METplus/parm/use_cases/met_tool_wrapper/Example/Example.conf /home/centos/METplus-4.0.0_Tutorial/tutorial.conf
03/22 23:13:14.213 metplus INFO: Log file: /home/centos/METplus-4.0.0_Tutorial/output/logs/metplus.log.20220322231314
03/22 23:13:14.213 metplus INFO: METplus Base: /home/centos/METplus
03/22 23:13:14.214 metplus INFO: Final Conf: /home/centos/METplus-4.0.0_Tutorial/output/metplus_final.conf
03/22 23:13:14.214 metplus INFO: Config Input: /home/centos/METplus/parm/metplus_config/defaults.conf
03/22 23:13:14.214 metplus INFO: Config Input: /home/centos/METplus/parm/use_cases/met_tool_wrapper/Example/Example.conf
03/22 23:13:14.214 metplus INFO: Config Input: /home/centos/METplus-4.0.0_Tutorial/tutorial.conf
```

**Comment by georgemccabe** [comment, marked as answer]
Upon further investigation, the AMI used here is not the version used for the METplus Training Series. This environment was set up by copying a defaults.conf file that was saved from METplus 4.0.0 with a few values changed into the cloned repository. This was creating the state where an old defaults.conf file was being used that is not compatible with METplus 4.1.0. I advised the developer who created this environment to use sed commands to replace values in the defaults.conf instead of copying an old file that may need to be updated over time.
